language-plutus-core/src/Language/PlutusCore/Normalize.hs

```diff
@@ -21,23 +25,33 @@ newtype TypeEnv tyname = TypeEnv
     { unTypeEnv :: IntMap (NormalizedType tyname ())
     }
 
-type NormalizeTypeM tyname = ReaderT (TypeEnv tyname) Quote
+type NormalizeTypeM tyname a = StateT GasInit (ReaderT (TypeEnv tyname) (ExceptT (TypeError a) Quote))
+
+type GasInit = Maybe Natural
```

Comment:
You don't actually use this in `PlutusCore`.

Reply:
What do you mean?